From: Simon McVittie <smcv@debian.org>
Date: Mon, 30 Aug 2021 10:54:43 +0100
Subject: compose: Update sequences from libX11 1.7.2

This adds support for sequences like <Compose>,G,u -> capital G with
breve. Previously, only a capital U was accepted for E, G, I and O
(but a lower-case u was accepted for A and U for some reason).

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3902
Applied-upstream: 4.5.0
---
 gtk/compose/gtkcomposedata.h          |  2 +-
 testsuite/gtk/compose/system.expected | 14 ++++++++++++--
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/gtk/compose/gtkcomposedata.h b/gtk/compose/gtkcomposedata.h
index 4426962..09b0263 100644
--- a/gtk/compose/gtkcomposedata.h
+++ b/gtk/compose/gtkcomposedata.h
@@ -3,7 +3,7 @@
 
 #define MAX_SEQ_LEN 5
 #define N_INDEX_SIZE 30
-#define DATA_SIZE 16447
+#define DATA_SIZE 16477
 #define N_CHARS 1241
 
 #endif
diff --git a/testsuite/gtk/compose/system.expected b/testsuite/gtk/compose/system.expected
index c807827..93c51eb 100644
--- a/testsuite/gtk/compose/system.expected
+++ b/testsuite/gtk/compose/system.expected
@@ -1,7 +1,7 @@
-# n_sequences: 4874
+# n_sequences: 4884
 # max_seq_len: 5
 # n_index_size: 30
-# data_size: 16447
+# data_size: 16477
 # n_chars: 1241
 <U7ae> <U7e9> : "ΐ" # U390
 <U7ae> <U7f5> : "ΰ" # U3b0
@@ -1959,6 +1959,7 @@
 <Uff20> <U47> <U2e> : "Ġ" # U120
 <Uff20> <U47> <U54> : ">" # U3e
 <Uff20> <U47> <U55> : "Ğ" # U11e
+<Uff20> <U47> <U75> : "Ğ" # U11e
 <Uff20> <U47> <U1a2> : "Ğ" # U11e
 <Uff20> <U48> <U2c> : "Ḩ" # U1e28
 <Uff20> <U49> <U22> : "Ï" # Ucf
@@ -2512,6 +2513,7 @@
 <Uff20> <U67> <U2e> : "ġ" # U121
 <Uff20> <U67> <U55> : "ğ" # U11f
 <Uff20> <U67> <U74> : ">" # U3e
+<Uff20> <U67> <U75> : "ğ" # U11f
 <Uff20> <U67> <U1a2> : "ğ" # U11f
 <Uff20> <U68> <U2c> : "ḩ" # U1e29
 <Uff20> <U69> <U22> : "ï" # Uef
@@ -2611,11 +2613,19 @@
 <Uff20> <U75> <U3b> : "ų" # U173
 <Uff20> <U75> <U3e> : "û" # Ufb
 <Uff20> <U75> <U41> : "Ă" # U102
+<Uff20> <U75> <U45> : "Ĕ" # U114
+<Uff20> <U75> <U47> : "Ğ" # U11e
+<Uff20> <U75> <U49> : "Ĭ" # U12c
+<Uff20> <U75> <U4f> : "Ŏ" # U14e
 <Uff20> <U75> <U55> : "Ŭ" # U16c
 <Uff20> <U75> <U5e> : "û" # Ufb
 <Uff20> <U75> <U5f> : "ū" # U16b
 <Uff20> <U75> <U60> : "ù" # Uf9
 <Uff20> <U75> <U61> : "ă" # U103
+<Uff20> <U75> <U65> : "ĕ" # U115
+<Uff20> <U75> <U67> : "ğ" # U11f
+<Uff20> <U75> <U69> : "ĭ" # U12d
+<Uff20> <U75> <U6f> : "ŏ" # U14f
 <Uff20> <U75> <U75> : "ŭ" # U16d
 <Uff20> <U75> <U7e> : "ũ" # U169
 <Uff20> <U75> <Ua8> : "ü" # Ufc
